homeassistant/components/volvooncall/binary_sensor.py

```diff
@@ -24,16 +24,23 @@ async def async_setup_platform(
 class VolvoSensor(VolvoEntity, BinarySensorEntity):
     """Representation of a Volvo sensor."""
 
+    def __init__(
+        self,
+        coordinator: VolvoUpdateCoordinator,
+        vin: str,
+        component: str,
+        attribute: str,
+        slug_attr: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(vin, component, attribute, slug_attr, coordinator)
+
+        if self.instrument.device_class in DEVICE_CLASSES:
```

Comment:
We can use the voluptuous schema to get the correct enum instead.

https://github.com/home-assistant/core/blob/a0adfb9e62564615c1de8ea619634ddc3e102936/homeassistant/components/binary_sensor/__init__.py#L119

Reply:
I'm not sure that I did this in the most idiomatic way, please let me know if you prefer a different approach.